packages/components/src/select-control/index.tsx

```diff
@@ -26,6 +26,22 @@ function useUniqueId( idProp?: string ) {
 	return idProp || id;
 }
 
+function SelectOptions( {
+	options,
+}: {
+	options: NonNullable< SelectControlProps[ 'options' ] >;
+} ) {
+	return options.map( ( { id, label, value, ...optionProps }, index ) => {
+		const key = id || `${ label }-${ value }-${ index }`;
```

Comment:
As a `key`, `${ label }-${ value }-${ index }` seems functionally equivalent to `index` 🤔 Doesn't seem particularly intentional, either (#6122).  I'm not touching this in this PR though.